### PR TITLE
add HEAD method for dry-run

### PR DESCRIPTION
**Pull request by jameslounds**
As of turbo 1.5.2, the --dry-run option checks the cache for the artefacts.

This PR lets the worker respond to the HEAD http method

### CODE REVIEW

**Comment by cometkim** [comment]
Can you clean up your fomatter's changes?

**Comment by jameslounds** [comment]
can you provide a prettier/eslint config for me to run against?

**Comment by jameslounds** [comment]
I've added a `prettierrc` file, and run it against the code.

I've got another branch where I'm using R2 since I have some artefacts larger than 1mb.
Should I open another PR to your main branch, or do you think it's best to have this on another branch?

**Comment by cometkim** [comment]
No just please change only relevant. you can use "Save without formatting" if you are on VSCode.

**Comment by jameslounds** [comment]
Consider this as two features then. I’ve added the HEAD method, and initialised prettier. This will avoid this exact issue should anyone else make a PR

File: src/worker.ts [hunk 1]
```diff
@@ -29,13 +29,22 @@ addEventListener('fetch', event => {
 
   switch (request.method) {
     case 'GET': {
+      // prettier-ignore
+      return void event.respondWith(
+        getArtifact(hash),
+      );
+    }
+
+    case 'HEAD': {
```
Comment (cometkim) [inline src/worker.ts:38]: Response to the `HEAD` method shouldn't have the body, only headers

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "tabWidth": 2,
+  "useTabs": false,
+  "quoteProps": "as-needed",
+  "singleQuote": true,
+  "trailingComma": "all",
+  "bracketSpacing": true,
+  "printWidth": 65,
+  "arrowParens": "avoid"
+}
```
Comment (cometkim) [inline .prettierrc:10]: Please remove irrelevant

File: src/worker.ts [hunk 3]
```diff
@@ -18,7 +18,7 @@ addEventListener('fetch', event => {
       new Response(null, { status: 404 }),
     );
   }
-
+  // prettier-ignore
```
Comment (cometkim) [inline src/worker.ts:21]:
```suggestion
```

File: src/worker.ts [hunk 4]
```diff
@@ -29,13 +29,22 @@ addEventListener('fetch', event => {
 
   switch (request.method) {
     case 'GET': {
+      // prettier-ignore
```
Comment (cometkim) [inline src/worker.ts:32]:
```suggestion
```

File: src/worker.ts [hunk 5]
```diff
@@ -29,13 +29,22 @@ addEventListener('fetch', event => {
...
+      return void event.respondWith(
+        getArtifact(hash),
+      );
+    }
+
+    case 'HEAD': {
+      // prettier-ignore
```
Comment (cometkim) [inline src/worker.ts:39]:
```suggestion
```

File: src/worker.ts [hunk 6]
```diff
@@ -29,13 +29,22 @@ addEventListener('fetch', event => {
...
       return void event.respondWith(
         getArtifact(hash),
       );
     }
 
     case 'PUT': {
       if (!request.body) {
+        // prettier-ignore
```
Comment (cometkim) [inline src/worker.ts:47]:
```suggestion
```

File: src/worker.ts [hunk 7]
```diff
@@ -53,6 +62,7 @@ addEventListener('fetch', event => {
   }
 });
 
+// prettier-ignore
```
Comment (cometkim) [inline src/worker.ts:65]:
```suggestion
```

File: src/worker.ts [hunk 8]
```diff
@@ -62,7 +72,8 @@ async function getArtifact(hash: string): Promise<Response> {
   }
 }
 
-async function putArtifact(hash: string, artifact: ReadableStream): Promise<Response> {
+// prettier-ignore
+async function putArtifact(hash: string, artifact: ReadableStream,): Promise<Response> {
```
Comment (cometkim) [inline src/worker.ts:76]:
```suggestion
async function putArtifact(hash: string, artifact: ReadableStream): Promise<Response> {
```

**Review by cometkim — CHANGES_REQUESTED** [review]

**Pull request closed without merging**